pkg/segment/aggregations/segaggs_test.go

```diff
@@ -2605,111 +2607,218 @@ func Test_findSpan(t *testing.T) {
+			result, exist, err := PerformNoWindowStreamStatsOnSingleFunc(ssOption, ssResults, measureAgg, value, true)
 			ssOption.NumProcessedRecords++
 			assert.Nil(t, err)
 			if !ssOption.Current {
 				if j == 0 {
 					assert.False(t, exist)
-					assert.Equal(t, defaultVal, result)
+					assert.True(t, compareValues(utils.CValueEnclosure{}, result))
 				} else {
 					assert.True(t, exist)
-					assert.Equal(t, expectedValues[i][j-1], result)
+					assert.True(t, compareValues(expectedValues[i][j-1], result))
 				}
 			} else {
 				assert.Equal(t, 0, ssResults.Window.Len())
 				assert.Equal(t, 0, ssResults.SecondaryWindow.Len())
-				if measureFunc == utils.Avg {
-					assert.Equal(t, expectedValues[i][j], result)
-					assert.Equal(t, expectedValues2[i][j], ssResults.CurrResult)
-				} else if measureFunc == utils.Range {
-					assert.Equal(t, expectedValues[i][j], ssResults.CurrResult)
-					assert.Equal(t, expectedValues2[i][j], ssResults.RangeStat.Max)
-					assert.Equal(t, expectedValues3[i][j], ssResults.RangeStat.Min)
+				if measureAgg.MeasureFunc == utils.Avg {
+					assert.True(t, compareValues(expectedValues[i][j], result))
+					assert.True(t, compareValues(expectedValues2[i][j], ssResults.CurrResult))
+				} else if measureAgg.MeasureFunc == utils.Range {
+					assert.True(t, compareValues(expectedValues[i][j], ssResults.CurrResult))
+					assert.True(t, compareValues(expectedValues2[i][j], utils.CValueEnclosure{Dtype: utils.SS_DT_FLOAT, CVal: ssResults.RangeStat.Max}))
+					assert.True(t, compareValues(expectedValues3[i][j], utils.CValueEnclosure{Dtype: utils.SS_DT_FLOAT, CVal: ssResults.RangeStat.Min}))
 				} else {
-					assert.Equal(t, expectedValues[i][j], ssResults.CurrResult)
+					assert.True(t, compareValues(expectedValues[i][j], ssResults.CurrResult))
 				}
 			}
 		}
 	}
 }
 
+func NoWindowStreamStatsHelperTest2(t *testing.T, values []utils.CValueEnclosure, ssOption *structs.StreamStatsOptions, measureAggs []*structs.MeasureAggregator, expectedValues [][]utils.CValueEnclosure) {
+
+	for i, measureAgg := range measureAggs {
+		ssResults := InitRunningStreamStatsResults(measureAgg.MeasureFunc)
+		ssOption.NumProcessedRecords = 0
+		for j, value := range values {
+			result, exist, err := PerformNoWindowStreamStatsOnSingleFunc(ssOption, ssResults, measureAgg, value, true)
+			ssOption.NumProcessedRecords++
+			assert.Nil(t, err)
+			if !ssOption.Current {
+				if j == 0 {
+					assert.False(t, exist)
+					assert.True(t, compareValues(utils.CValueEnclosure{}, result))
+				} else {
+					assert.True(t, compareValues(expectedValues[i][j-1], result))
+				}
+			} else {
+				assert.Equal(t, 0, ssResults.Window.Len())
+				assert.Equal(t, 0, ssResults.SecondaryWindow.Len())
+				assert.True(t, compareValues(expectedValues[i][j], result))
+			}
+		}
+	}
+}
+
+func getMeasureAgg(measureFuncs []utils.AggregateFunctions) []*structs.MeasureAggregator {
```

Comment:
nit: since you're returning a list:
```suggestion
func getMeasureAggs(measureFuncs []utils.AggregateFunctions) []*structs.MeasureAggregator {
```